Implement animated carousel in hero section

maribelferreira opened the pull request:
Enhancements include:
- Added brand images (Microsoft, XKCD, Gandalf, BBC, Chewbacca)
- Implemented an infinite horizontal scrolling carousel in the Hero component
- Adjusted responsive styling for logo display
- Updated layout and padding in several components for consistency

maribelferreira committed:
Implement animated logo carousel in Hero section

Enhancements include:
- Added brand logo images (Microsoft, XKCD, Gandalf, BBC, Chewbacca)
- Implemented an infinite horizontal scrolling carousel in the Hero component
- Adjusted responsive styling for logo display
- Updated layout and padding in several components for consistency

maribelferreira committed:
Enhance Hero section with new logos and improved carousel animation

- Added Spotify, Winter, and Audio waveform assets
- Updated carousel to include new logos and improve scrolling animation
- Adjusted layout and styling for better visual presentation

maribelferreira committed:
Refine Founder and Hero section layouts and responsiveness

- Adjusted Founder section image layout with centered and responsive positioning
- Updated Hero section carousel with improved spacing and mobile-friendly styles
- Added interactive carousel pause functionality on click/tap
- Optimized image and logo sizing for better visual presentation

maribelferreira committed:
Merge branch 'main' into carousel

gnarea commented on src/assets/spotify.png:
Sorry, I know this was a pain to create, but can you make it 2X slower? I think it's too fast, and too distracting.

gnarea commented on src/components/Homepage/Hero.astro:
```diff
@@ -1,27 +1,108 @@
+            <Image src={bbcImg} alt="BBC" class="h-30 w-auto object-contain" />
+            <Image src={winterImg} alt="Winter" class="h-24 w-auto object-contain" />
+          </div>
+          <div class="flex gap-8 sm:gap-16 items-center min-w-max">
+            <Image src={xkcdImg} alt="XKCD" class="h-24 w-auto object-contain" />
+            <Image src={microsoftImg} alt="Microsoft" class="h-24 w-auto object-contain" />
+            <div class="relative">
+              <Image src={spotifyImg} alt="Spotify" class="h-24 w-auto object-contain" />
+              <Image
+                src={audioGif}
+                alt="Waveform gif"
+                class="absolute top-1/2 left-1/2 -translate-x-[53%] -translate-y-1/2"
+              />
+            </div>
+            <Image src={gandalfImg} alt="Gandalf" class="h-24 w-auto object-contain" />
+            <Image src={bbcImg} alt="BBC" class="h-30 w-auto object-contain" />
+            <Image src={chewbaccaImg} alt="Chewbacca" class="h-24 w-auto object-contain" />
+            <Image src={winterImg} alt="Winter" class="h-24 w-auto object-contain" />
+          </div>
```
I think it's a very brittle hack to duplicate the carousel items. With the right code, it should just be able to loop indefinitely.

gnarea reviewed (changes requested):
- Desktop: I think it should stop when I hover over.
- Desktop: I've got a feeling that it may be so fast, that it's distracting the reader from reader the hero. Pushing it a bit further down the hero text might help too.
- I think the transition when it restarts is too abrupt.

maribelferreira committed:
Enhance Hero section carousel interaction and animation

- Modify carousel scroll animation duration from 20s to 30s
- Add hover-based pause functionality for desktop devices
- Maintain existing click/tap pause behavior for mobile interactions

maribelferreira committed:
Refine Hero section carousel with improved layout and scrolling

- Adjust carousel image order and spacing
- Add duplicate image set for seamless infinite scrolling
- Update tap functionality comment for clarity

maribelferreira committed:
Refactor Hero section carousel into reusable Carousel component

- Extract carousel logic into separate Carousel and CarouselImage components
- Replace static image carousel with dynamic Carousel component
- Update Spotify image to use animated GIF
- Simplify Hero section markup by delegating carousel rendering

gnarea commented on src/components/Homepage/ImageCarousel.astro:
This isn't used

gnarea reviewed (changes requested):
CI is breaking.

maribelferreira committed:
Remove unused ImageCarousel component

- Delete empty ImageCarousel.astro file
- Minor cleanup of CarouselImage component by removing empty line
